### DISCUSSION TITLE
Argon2 on Vercel

### DISCUSSION THREAD

**Discussion by riculum** [vercel/next.js#65978] [Help]
### Summary

I am building a Next.js app and plan to deploy it on Vercel. It runs fine locally, but when I deploy it to Vercel, I encounter a 500 error. The log shows the following message:
```
⨯ Error: No native build was found for platform=linux arch=x64 runtime=node abi=115 uv=1 libc=glibc node=20.12.0
    loaded from: /var/task/node_modules/argon2

    at load.resolve.load.path (/var/task/node_modules/node-gyp-build/node-gyp-build.js:60:9)
    at load (/var/task/node_modules/node-gyp-build/node-gyp-build.js:22:30)
    at Object.<anonymous> (/var/task/node_modules/argon2/argon2.cjs:7:32)
    at Module._compile (node:internal/modules/cjs/loader:1369:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1427:10)
    at Module.load (node:internal/modules/cjs/loader:1206:32)
    at Module._load (node:internal/modules/cjs/loader:1022:12)
    at /opt/rust/nodejs.js:1:11508
    at Function.Qt (/opt/rust/nodejs.js:1:11878)
    at Q.e.<computed>.K._load (/opt/rust/nodejs.js:1:11478) {
  page: '/check'
}
```

How can I resolve this issue? Do I have to switch to bcrypt?

**Comment by zachlankton** [comment]
this seems to be recent, I've been using argon2 for quite some time now, and today am having this same issue.  

Same issue occurs across my m3 mac and my x86 linux server

**Reply by nyaarium** [reply]
I just tried installing and building with argon2 just now, and got the same issue. Do you know if the issue stops with certain versions of Next or Argon?

**Comment by avarayr** [comment]
Solution: downgrade to `0.31.2` and pin it

`npm i --save-exact argon2@0.31.2`

**Reply by Vandivier** [reply]
this worked for me without downgrading:
https://github.com/vercel/next.js/discussions/65978#discussioncomment-10990903

**Comment by kkoukos** [comment]
Any new fixes apart from downgrading?

**Comment by NazmusSayad** [comment]
Any update guys? Facing the same problem 😥

**Reply by jin-current** [reply]
Someone in my team fixed this by adding the below to `next.config.js`

```
module.exports = {
  reactStrictMode: true,
  transpilePackages: ["ui"],
  experimental: {
    serverComponentsExternalPackages: ["@react-pdf/renderer"],
    ...

    // Fixes issues with argon2 and vercel
    // - https://github.com/vercel/next.js/discussions/65978
    // - https://github.com/ranisalt/node-argon2/issues/402#issuecomment-2073016949
    outputFileTracingIncludes: {
      "/": ["../../node_modules/argon2/prebuilds/linux-x64/*"],
    },
  },
```

**Reply by NazmusSayad** [reply]
I was searching on Stack Overflow when I came across this.

To install npm packages on Vercel, use the `npm install && npm rebuild argon2 --build-from-source`. You can configure this using vercel.json or similar methods.

And it just works.

`vercel.json`

```json
{
  "installCommand": "npm install && npm rebuild argon2 --build-from-source"
}
```
